server/app/controllers/applicant/ApplicantRoutes.java

```diff
@@ -82,4 +82,21 @@ public Call show(CiviFormProfile profile, long applicantId, long programId) {
           String.valueOf(programId));
     }
   }
+
+  /**
+   * Returns the route corresponding to the applicant edit action.
+   *
+   * @param profile - Profile corresponding to the logged-in user (applicant or TI).
+   * @param applicantId - ID of applicant for whom the action should be performed.
+   * @param programId - ID of program to edit
+   * @return Route for the applicant action
+   */
+  public Call edit(CiviFormProfile profile, long applicantId, long programId) {
```

Comment:
**suggestion:** I _think_ we can hypothetically inject profileUtils and get the civiform profile in this file instead of having to add it to the class of each caller. That may not be optimal for another reason, but was curious if you had considered that option.

Reply:
You are right that we can inject `ProfileUtils`, but then we need the current `request` to call `.currentUserProfile()`: https://github.com/civiform/civiform/blob/0b6c86265fd5b42861e22fb5a76631a4a2c9b792/server/app/auth/ProfileUtils.java#L33

So it doesn't seem like a net win, just trading one parameter for another. In many places, we've already extracted the profile from the request, so I figured we can re-use that rather than repeating the extraction.

Reply:
Ah true! That makes sense to me too